packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/dropNode/useDropNode.ts

```diff
@@ -95,27 +95,32 @@ const useDropNodeMutation = () => {
     }
   }, [dropNodeData, dropNodeError]);
 
-  const invokeMutation = (
-    droppedNode: Node,
-    targetElementId: string | null,
-    dropPosition: XYPosition,
-    onDragCancelled: (node: Node) => void
-  ): void => {
-    const input: GQLDropNodeInput = {
-      id: crypto.randomUUID(),
-      editingContextId,
-      representationId: diagramId,
-      droppedElementId: droppedNode.id,
-      targetElementId,
-      x: dropPosition.x,
-      y: dropPosition.y,
-    };
-    dropMutation({ variables: { input } }).then((result) => {
-      if (result.data?.dropNode && isErrorPayload(result.data?.dropNode)) {
-        onDragCancelled(droppedNode);
+  const invokeMutation = useCallback(
```

Comment:
Why not prevent any move at all instead of just the mutation?

Reply:
Moves cannot be made.
But since this is API, this hook could be called from anywhere